### PR TITLE
Allow `defs` in the `svg` to be overwritten.

### PR DESCRIPTION
**Pull request by woutervdub**
Implements #3819

### CODE REVIEW

**Review by eriese — CHANGES_REQUESTED** [review]
Great work!

My one nitpick is that I think it'd be ideal if all the snippets were treated the way you've done `buttons`, i.e. each individual `snippet` gets added to `defs` via its own `id`, rather than all being grouped by function

**Comment by woutervdub** [comment]
The buttons one is different because there were multiple `defs` in the original. Each of those `defs` is itself made up from separate `id`s. So the `cutOnFold` is really the same as `snaps`, whereas `cutOnFold` is in its own mjs source file, while `snaps` is combined in the buttons mjs source file.

**Comment by woutervdub** [comment]
I guess I broke a bunch of tests.

File: packages/core/tests/svg.test.mjs [hunk 1]
```diff
@@ -42,7 +43,7 @@ describe('Svg', () => {
     expect(svg.freeId).to.equal(0)
     expect(svg.body).to.equal('')
     expect(svg.style).to.equal('')
-    expect(svg.defs).to.equal('')
+    expect(svg.defs instanceof Defs).to.equal(true)
```
Comment (eriese) [inline packages/core/tests/svg.test.mjs:46]: there's a handy assertion you can use here that's a little more readable. `expect(svg.defs).to.be.an.instanceof(Defs)`. It'll throw a better error

File: plugins/plugin-annotations/tests/buttons.test.mjs [hunk 2]
```diff
@@ -11,7 +11,7 @@ pattern.draft().render()
 describe('Buttons Plugin Test', () => {
   for (const snippet of ['button', 'buttonhole', 'snap-stud', 'snap-socket']) {
     it(`Should add the ${snippet} snippet to defs`, () => {
-      expect(pattern.svg.defs.indexOf(`<g id="${snippet}">`)).to.not.equal(-1)
+      expect(pattern.svg.defs.render().indexOf(`<g id="${snippet}">`)).to.not.equal(-1)
```
Comment (eriese) [inline plugins/plugin-annotations/tests/buttons.test.mjs:14]: change this to `expect(pattern.svg.defs).to.have.property(snippet)`

File: plugins/plugin-annotations/tests/logo.test.mjs [hunk 3]
```diff
@@ -9,7 +9,7 @@ describe('Logo Plugin Tests', () => {
     const Pattern = new Design()
     const pattern = new Pattern().use(annotationsPlugin)
     pattern.draft().render()
-    expect(pattern.svg.defs).to.contain(
+    expect(pattern.svg.defs.render()).to.contain(
```
Comment (eriese) [inline plugins/plugin-annotations/tests/logo.test.mjs:12]: same here. `expect(pattern.svg.defs).to.have.property('logo')`

File: plugins/plugin-annotations/tests/notches.test.mjs [hunk 4]
```diff
@@ -17,11 +17,11 @@ pattern.draft().render()
 
 describe('Notches Plugin Test', () => {
   it(`Should add the snippets to defs`, () => {
-    expect(pattern.svg.defs).to.contain('<g id="notch">')
+    expect(pattern.svg.defs.render()).to.contain('<g id="notch">')
   })
 
   it(`Should add the notches snippet to defs`, () => {
-    expect(pattern.svg.defs.indexOf(`<g id="notch">`)).to.not.equal(-1)
+    expect(pattern.svg.defs.render().indexOf(`<g id="notch">`)).to.not.equal(-1)
```
Comment (eriese) [inline plugins/plugin-annotations/tests/notches.test.mjs:24]: These as well

**Review by eriese — DISMISSED** [review]

**Comment by eriese** [comment]
It's looking really good. Only thing is test coverage. I'd say if you haven't seen a use case for `Defs.prototype.clone()` you could get rid of it and save yourself a test

**Review by eriese — APPROVED** [review]

**Comment by BenJamesBen** [comment]
I do not know if these are related to this PR's changes or if they simply might be due to this PR being based on older code, but I notice that in Cutting Layout:
1. Parts no longer align with grain when the grainline is toggled to horizontal.
2. Cutonfold parts are no longer mirrored/splayed.
3. Some parts no longer show via `complete`, like Carlita's `carlton.back`.

**Comment by eriese** [comment]
I think that's because this branch is 230 commits behind `develop`, and those things were broken in the move to `plugin-annotations`, but I think it's probably best for @woutervdub to merge in develop to find out

**Comment by eriese** [comment]
@joostdecock I'm good with this, but don't want to merge it without your approval because I think you had a point of view about how this should be implemented

**Comment by joostdecock** [comment]
I had a small remark but it's not something @woutervdub did, just something I noticed while reviewing this.
So I've merged this (great work) and created: #3965 3965

**Pull request merged**